py/dml/provisional.py

```diff
@@ -81,6 +83,57 @@ class explicit_param_decls(ProvisionalFeature):
     short = "Require := syntax for defining new params"
     stable = True
 
+
+@feature
+class simics_util_vect(ProvisionalFeature):
+    '''<a id="simics_util_vect"/>
+    This feature enables the `vect` type, based on the
+    `VECT` macro from the Simics C API (`simics/util/vect.h`).
+
+    This is a simple wrapping that behaves inconsistently in many
+    ways, and we plan to eventually introduce a cleaner mechanism for
+    vectors; the `simics_util_vect` is supported as an interim solution until we
+    have that in place.
+
+    The syntax is `BASETYPE vect`, e.g. `typedef int vect int_vect_t;`
+    to define a type for vectors of the `int` type.
+
+    Some caveats:
+
+    * `vect` types typically need to be `typedef`:ed before they are
+      used.  This is because `int vect` is blindly expanded into
+      `VECT(int)` in C, which in turn expands into a `struct`
+      definition, meaning that saying `VECT(int)` twice yields two
+      incompatible types. This means, for instance, that `typeof` in
+      DML doesn't work properly for `vect` types unless `typedef`:ed
+
+    * Importing `"internal.dml"` exposes various C macros from
+      `vect.h` to DML: `VINIT`, `VELEMSIZE`, `VRESIZE`,
+      `VRESIZE_FREE`, `VADD`, `VREMOVE`, `VDELETE_ORDER`, `VINSERT`,
+      `VSETLAST`, `VLEN`, `VVEC`, `VGROW`, `VSHRINK`, `VFREE`,
+      `VTRUNCATE`, `VCLEAR`, and `VCOPY`.
+
+    * DML natively supports indexing syntax, which is translated to
+      `VGET` (or `VSET` for assignment). For instance:
+      ```
+      typedef int vect int_vect_t;
+      method first_element(int_vect_t v) -> (int) {
+          assert VLEN(v) > 0;
+          return v[0];
+      }
+      ```
+
+    Enabling the `simics_util_vect` feature in a file only affects
+    the `vect` declarations in that file.
+
+    When the `simics_util_vect` feature is disabled, usage of `vect` is an
+    error unless the [`experimental_vect` compatibility
+    feature](deprecations-auto.html#experimental_vect) is enabled.
+    '''
+    short = "Allow vect syntax based on the VECT macro"
```

Comment:
I'm inclined to bikeshed this short desc a bit, for multiple reasons:
* To make it more obvious as to what the syntax actually is and does (it's possibly to read this as the DML syntax also being `VECT`)
* To explicitly state the connection to the Simics API
* And to somewhat future-proof the formulation against the introduction of RAII vectors

My suggestion would be:
> Enable 'vect' type syntax that compile to usages of the Simics 'VECT' macro

However, I don't insist on any on this, and this is not blocking.

Reply:
I prefer to leave disambiguation to the full docs. I guess most of the time when you look at the list it's one of the other features you are looking for, which means the main use case for the short desc is to quickly convince you that this is *not* the thing you are looking for. So better to be terse and simple. If you don't know what the VECT macro is about, then you don't use the feature and the short desc is not for you.